network/secondary_endpoint_client_linux.go

```diff
@@ -0,0 +1,150 @@
+package network
+
+import (
+	"net"
+
+	"github.com/pkg/errors"
+
+	"github.com/Azure/azure-container-networking/log"
+	"github.com/Azure/azure-container-networking/netio"
+	"github.com/Azure/azure-container-networking/netlink"
+	"github.com/Azure/azure-container-networking/network/networkutils"
+	"github.com/Azure/azure-container-networking/platform"
+)
+
+var errorSecondaryEndpointClient = errors.New("SecondaryEndpointClient Error")
+
+func newErrorSecondaryEndpointClient(err error) error {
+	return errors.Wrapf(err, "%s", errorSecondaryEndpointClient)
+}
+
+type SecondaryEndpointClient struct {
+	netlink        netlink.NetlinkInterface
+	netioshim      netio.NetIOInterface
+	plClient       platform.ExecClient
+	netUtilsClient networkutils.NetworkUtils
+	ep             *endpoint
+}
+
+func NewSecondaryEndpointClient(
+	nl netlink.NetlinkInterface,
+	plc platform.ExecClient,
+	endpoint *endpoint,
+) *SecondaryEndpointClient {
+	client := &SecondaryEndpointClient{
+		netlink:        nl,
+		netioshim:      &netio.NetIO{},
+		plClient:       plc,
+		netUtilsClient: networkutils.NewNetworkUtils(nl, plc),
+		ep:             endpoint,
+	}
+
+	return client
+}
+
+func (client *SecondaryEndpointClient) AddEndpoints(epInfo *EndpointInfo) error {
+	iface, err := client.netioshim.GetNetworkInterfaceByMac(epInfo.MacAddress)
+	if err != nil {
+		return newErrorSecondaryEndpointClient(err)
+	}
+
+	epInfo.IfName = iface.Name
+	if _, exists := client.ep.SecondaryInterfaces[iface.Name]; exists {
+		return newErrorSecondaryEndpointClient(errors.New(iface.Name + " already exists"))
+	}
+	client.ep.SecondaryInterfaces[iface.Name] = &InterfaceInfo{
+		Name:               iface.Name,
+		MacAddress:         epInfo.MacAddress,
+		IPAddress:          epInfo.IPAddresses,
+		AddressType:        epInfo.AddressType,
+		IsDefaultInterface: epInfo.IsDefaultInterface,
+	}
+
+	return nil
+}
+
+func (client *SecondaryEndpointClient) AddEndpointRules(_ *EndpointInfo) error {
+	return nil
+}
+
+func (client *SecondaryEndpointClient) DeleteEndpointRules(ep *endpoint) {
+	// ip route del <multitenant ip> dev <ethX>
+	// Deleting the route set up for routing the incoming packets to pod
+	for ifName, ifInfo := range ep.SecondaryInterfaces {
+		log.Printf("[net] deleting routes on %s: %+v", ifName, ifInfo.Routes)
+		if err := deleteRoutes(client.netlink, client.netioshim, ifName, ifInfo.Routes); err != nil {
```

Comment:
who adds this route? can you point me to function that adds this route?

Reply:
they're added in ConfigureContainerInterfacesAndRoutes

Reply:
isn't this route added in VM namespace? `DeleteEndpointRules` is called from VM namespace while `ConfigureContainerInterfacesAndRoutes` called in container namespace

Reply:
i checked `ConfigureContainerInterfacesAndRoutes`..i didnt see route getting added.. can you share link?

Reply:
yup they're added in container namespace so not sure if we actually need this part

Reply:
if not required, please remove it